pkg/providers/translation/upstream.go

```diff
@@ -166,10 +164,12 @@ func (t *translator) translateUpstreamV2beta3(ep *kube.Endpoint, namespace, name
 		return nil, err
 	}
 	ups.Nodes = nodes
+	ups.Name = apisixv1.ComposeUpstreamName(namespace, name, subset, port.IntVal)
```

Comment:
IMHO, `TranslateUpstreamConfigVxx` should take an upstream parameter and modify it directly, instead of generating a new one. Would you like to edit this?

Reply:
I agree that the configuration should be applied instead of generating new objects.